card/mod/05_standard/set/right/token.rb

```diff
@@ -3,3 +3,29 @@
 view :raw do |args|
   "Private data"
 end
+
+def validate! token
+  error =
+    case
+    when !real?           then [:token_not_found, 'no token found']
+    when expired?         then [:token_expired, 'expired token']
+    when content != token then [:incorrect_token, 'token mismatch']
+    end
+  errors.add *error if error
+end
+
+def expired?
+  !permanent? && updated_at <= term.ago
+end
+
+def permanent?
+  false
+end
```

Comment:
Haven't decided how to make a token permanent yet, but I was thinking to make it so you could add an +*expiration card to customize whether and when a token expires.